src/tpm2_dictionarylockout.c

```diff
@@ -0,0 +1,229 @@
+    };
+
+    int opt;
+    bool result;
+    while ((opt = getopt_long(argc, argv, "n:r:l:P:CS", long_options, NULL))
+            != -1) {
+        switch (opt) {
+        case 'C':
+            ctx->clear_lockout = true;
+            break;
+        case 'S':
+            ctx->setup_parameters = true;
+            break;
+        case 'P':
+            snprintf(ctx->lockout_passwd, sizeof(ctx->lockout_passwd), 
+                "%s", optarg);
+            if (!result) {
+                return false;
+            }
+            ctx->use_passwd = true;
+            break;
+        case 'n':
+            result = string_bytes_get_uint32(optarg, &ctx->max_tries);
+            if (!result) {
+                LOG_ERR("Could not convert max_tries to number, got: \"%s\"",
+                        optarg);
+                return false;
+            }
+            if (ctx->max_tries == 0) {
+                LOG_ERR("max_tries cannot be 0");
+                return false;
+            }
+            break;
+        case 'r':
+            result = string_bytes_get_uint32(optarg, &ctx->recovery_time);
+            if (!result) {
+                LOG_ERR("Could not convert recovery_time to number, got: \"%s\"",
+                        optarg);
+                return false;
+            }
+            break;
+        case 'l':
+            result = string_bytes_get_uint32(optarg, &ctx->lockout_recovery);
+            if (!result) {
+                LOG_ERR("Could not convert lockout_recovery to number, got: \"%s\"",
+                        optarg);
+                return false;
+            }
+            break;        
+        case ':':
+            LOG_ERR("Argument %c needs a value!\n", optopt);
+            return false;
+        case '?':
+            LOG_ERR("Unknown Argument: %c\n", optopt);
+            return false;
+        default:
+            LOG_ERR("?? getopt returned character code 0%o ??\n", opt);
+            return false;
+        }
+    }
+
+    if(ctx->clear_lockout==false && ctx->setup_parameters==false) {
+        LOG_ERR("Invalid operational input: Neither Setup nor Clear lockout requested.\n");
+        return false;
+    }
+
+    return true;
+}
+
+int execute_tool(int argc, char *argv[], char *envp[], common_opts_t *opts,
+        TSS2_SYS_CONTEXT *sapi_context) {
+
+    (void)opts;
+    (void)envp;
+
+    if(argc==1) {
+        showArgMismatch(argv[0]);
+        return -1;
+    }
+
+    dictionarylockout_ctx ctx = {
+            .max_tries = 0,
+            .recovery_time = 0,
+            .lockout_recovery = 0,
+            .clear_lockout = false,
+            .setup_parameters = false,
+            .use_passwd = true,
+            .sapi_context = sapi_context
+    };
+
+    bool result = init(argc, argv, &ctx);
+    if (!result) {
+        return 1;
+    }
+
+    result = dictionary_lockout_reset_and_parameter_setup(&ctx);
+    if(!result) {
+        return 2;
```

Comment:
return dictionary_lockout_reset_and_parameter_setup(&ctx) != true;

This picks up the tail call optimization and true != true is 0 and false != true is 1, so this normalizes our return values as well. Since were not documenting different return codes and were failing loudly via LOG_ERR() statements, lets just return 0 and 1 always.